src/Analyzer/Utils.cpp

```diff
@@ -975,4 +975,103 @@ QueryTreeNodePtr buildSubqueryToReadColumnsFromTableExpression(const QueryTreeNo
     return buildSubqueryToReadColumnsFromTableExpression(columns_to_select, table_node, context);
 }
 
+bool hasUnknownColumn(const QueryTreeNodePtr & node, QueryTreeNodePtr table_expression)
+{
+    QueryTreeNodes stack = { node };
+    while (!stack.empty())
+    {
+        auto current = stack.back();
+        stack.pop_back();
+
+        switch (current->getNodeType())
+        {
+            case QueryTreeNodeType::CONSTANT:
+                break;
+            case QueryTreeNodeType::COLUMN:
+            {
+                auto * column_node = current->as<ColumnNode>();
+                auto source = column_node->getColumnSourceOrNull();
+                if (source && table_expression && !source->isEqual(*table_expression))
```

Comment:
This is a bug and it leads to a `LOGICAL_ERROR`: https://github.com/ClickHouse/ClickHouse/issues/82092

1. `table_expression` should never be `nullptr`.
2. If `source` is `nullptr`, the function must return `true`, because the source table expression was replaced.